### PR TITLE
Add Super admin routes for user and orga management

### LINKED ISSUE #7148: Super admin routes for user and orga management
## Detailed Description
It would be very convenient to have the following routes available to super admins:
- Add an existing user to another organization (with user email/id, organization name)
- Create a new user that is automatically added to an organization (with names, email, password, organization name)
- Create a new organization (with organization name, owner user email/id)

## Context

* [ ] Specific to long-running jobs (set `jobsEnabled=true` in `application.conf`)
* [ ] Specific to webknossos.org (set `isDemoInstance=true` in `application.conf`)

### PR DESCRIPTION
**Pull request by frcroth**
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Create a new organization via POST api/organizations with sample2 user (This allows you to specify an existing user as admin) / use /auth/createOrganizationWithAdmin (This creates a new user as admin)
- Add user sample to this organization via POST api/organizations/:name/addUser
- Create a new user in this organization via POST auth/createUserInOrganization/:name
- Validate that all of this only works for super users.

### Issues:
- fixes #7148

------
(Please delete unneeded items, merge only when none are left open)
- [x] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [x] Considered [common edge cases](../blob/master/.github/common_edge_cases.md)

### CODE REVIEW

**Review by fm3 — CHANGES_REQUESTED** [review]
Thanks for having a look at this!

I think your version works well, but it could be simplified and be made more feature-complete by reusing some of the existing methods used during user signup/organization switching. I added some comments pointing to those. 

Maybe you could have a look at that? Feel free to reorganize some of those, if it makes things easier, none of this is set in stone :)

We might even alter the existing routes (of course then we would have to adapt the frontend).

File: app/controllers/OrganizationController.scala [hunk 1]
```diff
@@ -154,6 +172,22 @@ class OrganizationController @Inject()(organizationDAO: OrganizationDAO,
...
+
+  def addUser(organizationName: String): Action[AddUserParameters] =
+    sil.SecuredAction.async(validateJson[AddUserParameters]) { implicit request =>
+      for {
+        _ <- userService.assertIsSuperUser(request.identity._multiUser) ?~> "notAllowed" ~> FORBIDDEN
+        multiUserId <- userService.getMultiUserId(request.body.multiUserId, request.body.userEmail)
+        organization <- organizationDAO.findOneByName(organizationName) ?~> Messages("organization.notFound",
+                                                                                     organizationName) ~> NOT_FOUND
+        user <- userService.createUserInOrganization(multiUserId, organization._id, asOwner = false)
+      } yield Ok(user._id.toString)
```
[the remark this answers is not in the document]
Comment (frcroth) [reply app/controllers/OrganizationController.scala:188]: Just used the specification in the issue

[the remark this answers is not in the document]
Comment (fm3) [reply app/controllers/OrganizationController.scala:188]: Fair enough ;) However, if you’re ok with it, let’s just support email

File: conf/webknossos.latest.routes [hunk 2]
```diff
@@ -227,12 +228,14 @@ GET           /statistics/users
 
 # Organizations
 GET           /organizations                                                                    controllers.OrganizationController.list
+POST          /organizations                                                                    controllers.OrganizationController.create
```
[the remark this answers is not in the document]
Comment (frcroth) [reply conf/webknossos.latest.routes]: But this route always seems to create a new user? I understood that the new route should create an org with an existing user

[the remark this answers is not in the document]
Comment (fm3) [reply conf/webknossos.latest.routes]: Maybe I misunderstood the requirement then. @normanrz Could you clarify what you need here?

[the remark this answers is not in the document]
Comment (normanrz) [reply conf/webknossos.latest.routes]: > * Create a new organization (with organization name, owner user email/id)

I think what is missing is creating a new org for an existing user. For a new user, we can just use the signup.

File: app/controllers/OrganizationController.scala [hunk 3]
```diff
@@ -64,6 +65,23 @@ class OrganizationController @Inject()(organizationDAO: OrganizationDAO,
...
+    implicit val jsonFormat: OFormat[OrganizationCreationParameters] = Json.format[OrganizationCreationParameters]
+  }
+  def create: Action[OrganizationCreationParameters] =
+    sil.SecuredAction.async(validateJson[OrganizationCreationParameters]) { implicit request =>
+      for {
+        _ <- userService.assertIsSuperUser(request.identity._multiUser) ?~> "notAllowed" ~> FORBIDDEN
+        owner <- multiUserDAO.findOneByEmail(request.body.ownerEmail) ?~> "user.notFound"
+        org <- organizationService.createOrganization(request.body.organization, request.body.organizationDisplayName)
+        user <- userDAO.findFirstByMultiUser(owner._id)
+        _ <- userService.joinOrganization(user, org._id, autoActivate = true, isAdmin = true)
```
Comment (fm3) [inline app/controllers/OrganizationController.scala:81]:
```suggestion
        _ <- userService.joinOrganization(user, org._id, autoActivate = true, isAdmin = true, isOwner = true)
```
(should be propagated to the user’s isOrganizationOwner field)

File: app/models/user/UserService.scala [hunk 4]
```diff
@@ -81,6 +82,12 @@ class UserService @Inject()(conf: WkConf,
       _ <- bool2Fox(userBox.isEmpty) ?~> "organization.alreadyJoined"
     } yield ()
 
+  def assertIsSuperUser(multiUserId: ObjectId)(implicit ctx: DBAccessContext): Fox[Unit] =
+    for {
+      isSuperUser <- multiUserDAO.findOne(multiUserId).map(_.isSuperUser)
+      _ <- bool2Fox(isSuperUser)
+    } yield ()
```
Comment (fm3) [inline app/models/user/UserService.scala:89]:
```suggestion
Fox.assertTrue(multiUserDAO.findOne(multiUserId).map(_.isSuperUser))
```
(could you try that? I think that’s the same but maybe I’m overlooking something)

File: conf/messages [hunk 5]
```diff
@@ -63,6 +63,7 @@ user.notAuthorised=You are not authorized to view this resource. Please log in.
 user.id.notFound=We could not find a user id in the request.
 user.id.invalid=The provided user id is invalid.
 user.creation.failed=Failed to create user
+user.id.orEmailNecessary=Email or id needed to identify user.
```
Comment (fm3) [inline conf/messages:66]: probably unused now?

**Review by fm3 — COMMENTED** [review]
Almost there :)

**Pull request merged**